sourcegraph/sourcegraph
# Code Insights: Adjust routing and layouts for the landing page

## Issues
- Closes https://github.com/sourcegraph/sourcegraph/issues/30107
- Closes https://github.com/sourcegraph/sourcegraph/issues/30108

## Context

This PR adds a new routing system based on our [designs for the landing page](https://www.figma.com/file/EFgTP7gpdpQSEo1n4FUfsW/In-product-landing-page-WIP?node-id=1128%3A9814) Note that these new layouts and routing are available only if you have `codeInsightsLandingPage: true` in the `experimentalFeatures` in your setting file. 

## How to test
Since this PR changes our routing a lot we have to 
- Check all URLs and transitions between URLs for the old layouts `codeInsightsLandingPage: false` 
- Check all transitions for the new URL with  `codeInsightsLandingPage: true`. Note that our BE (if you're using stating doesn't know anything about the new experimental features flag that we added here. My suggestion is just to change the variable `codeInsightsLandingPage` in the `InsightsRouter.tsx` to true)
